### PR TITLE
Add Pod log for kubernetes agent

### PR DESCRIPTION
**Pull request by LinuxSuRen**
I don't know whether there will have a kubernetes environment, so I don't offer a test case.

### CODE REVIEW

**Comment by LinuxSuRen** [comment]
@carlossg Could you please help to review this PR? Thanks.

**Comment by LinuxSuRen** [comment]
@jenkinsci/cloud-native-sig Anyone could help me to review this? Thanks.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer.java [hunk 1]
```diff
@@ -40,6 +56,39 @@ public void taskCompletedWithProblems(Executor executor, Queue.Task task, long d
...
+
+        String namespace = StringUtils.defaultIfBlank(slave.getNamespace(), client.getNamespace());
+        Pod pod = client.pods().inNamespace(namespace).withName(getName()).get();
+
+        return pod.getSpec().getContainers();
+    }
+
+    public void doContainerLog(@QueryParameter String containerId,
+                               StaplerRequest req, StaplerResponse rsp) throws UnrecoverableKeyException, CertificateEncodingException, NoSuchAlgorithmException, KeyStoreException, IOException {
+        ByteBuffer outputStream = new ByteBuffer();
```
Comment (oleg-nenashev) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer.java]: No permission check here, the log becomes accessible to any user who is able to read the agent

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<!--
```
Comment (oleg-nenashev) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly]: This sidepanel is not referenced from anywhere

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer.java [hunk 3]
```diff
@@ -40,6 +56,39 @@ public void taskCompletedWithProblems(Executor executor, Queue.Task task, long d
         LOGGER.log(Level.FINE, " Computer " + this + " taskCompletedWithProblems");
     }
 
+    @Exported
+    public List<Container> getContainers() throws UnrecoverableKeyException, CertificateEncodingException, NoSuchAlgorithmException, KeyStoreException, IOException {
+        KubernetesSlave slave = getNode();
```
Comment (oleg-nenashev) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer.java]: permission checks are missing, not sure if READ permission should be enough to access such data

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer.java (hunk 1 above)
Comment (carlossg) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer.java]: this ?

    Jenkins.get().checkPermission(Jenkins.ADMINISTER);

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly (hunk 2 above)
Comment (LinuxSuRen) [reply src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly]: This file still working, it overrides from log.jelly.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer.java (hunk 3 above)
Comment (LinuxSuRen) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer.java]: Yes, the only administrator can get this data. Thanks.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer.java (hunk 1 above)
Comment (LinuxSuRen) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer.java]: Already add permission check.

**Comment by LinuxSuRen** [comment]
Does it have any other issues?

**Comment by LinuxSuRen** [comment]
ping @oleg-nenashev

**Pull request merged**